### PR TITLE
feat: introducing unit tests and general refactor

### PR DESCRIPTION
**Pull request by domenicoblanco**

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -3,20 +3,61 @@
...
+            print("id: " + chat_id)
+            url = f'https://api.telegram.org/bot{getenv("QDBotToken")}/sendMessage?chat_id={chat_id}&text={message}'
+
+            async with httpx.AsyncClient(http2=True) as client:
+                res = await client.post(url)
+                return res.json()
+
+
+async def main() -> None:
+    await make_request_to_telegram('https://duckduckgo.com/')
```
Comment (Valeri0p) [inline src/main.py]: Two arguments are needed here.
Also please use example.org from IANA, it has PING enabled.

File: src/main.py [hunk 2]
```diff
@@ -6,23 +6,33 @@
...
 
-#si, è copiata
+
+def check_ok(url: str) -> bool:
+    r = httpx.get(url)
+    #there are various status codes, duckduckgo returns 301; is there a keyword only for errors?
+    #return r.status_code == httpx.codes.OK
+    return r.status_code < 400
+
+
```
Comment (Valeri0p) [inline src/main.py:22]: This function is duplicated.

File: src/main.py (hunk 1 above)
Comment (domenicoblanco) [reply src/main.py]: Yeah, I didn't notice this change in the previous PR, it should be better to change some things

File: src/main.py (hunk 2 above)
Comment (domenicoblanco) [reply src/main.py:22]: My bad during the merge 😅

**Review by Valeri0p — APPROVED** [review]

**Pull request merged**